tests/SimplifyTest.elm

```diff
@@ -309,6 +309,297 @@ a = List.foldl f x << UniqueList.toList
+        ( _, Node [] _ ) ->
+            List.foldl f x << Set.toList
+
+        ( _, Node _ ({ foldl } :: _) ) ->
+            []
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold a set, you don't need to convert to a List"
+                            , details = [ "Using Set.foldl directly is meant for this exact purpose and will also be faster." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Set exposing (foldl)
+a =
+    case info of
+        ( _, Node [] _ ) ->
+            Set.foldl f x
+
+        ( _, Node _ ({ foldl } :: _) ) ->
+            []
+"""
+                        ]
+        , test "should qualify if imported and exposed but shadowed by let declaration argument" <|
+            \() ->
+                """module A exposing (..)
+import Set exposing (foldl)
+a =
+    let
+        doIt ( _, Node _ ({ foldl } :: _) ) =
+            List.foldl f x << Set.toList
+    in
+    doIt
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold a set, you don't need to convert to a List"
+                            , details = [ "Using Set.foldl directly is meant for this exact purpose and will also be faster." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Set exposing (foldl)
+a =
+    let
+        doIt ( _, Node _ ({ foldl } :: _) ) =
+            Set.foldl f x
+    in
+    doIt
+"""
+                        ]
+        , test "should not qualify if imported and exposed and same binding in argument of different let declaration" <|
+            \() ->
+                """module A exposing (..)
+import Set exposing (foldl)
+a =
+    let
+        doIt ( _, Node _ ({ foldl } :: _) ) =
+            ()
+        
+        doItBetter x =
+            List.foldl f x << Set.toList
+    in
+    doItBetter
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold a set, you don't need to convert to a List"
+                            , details = [ "Using Set.foldl directly is meant for this exact purpose and will also be faster." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Set exposing (foldl)
+a =
+    let
+        doIt ( _, Node _ ({ foldl } :: _) ) =
+            ()
+        
+        doItBetter x =
+            Set.foldl f x
```

Comment:
Should this not be `foldl f x`? `foldl` from `doIt` is not accessible here.

Reply:
right